src/Language/PureScript/Docs/Prim.hs

```diff
@@ -419,6 +420,14 @@ symbolCons = primClassOf (P.primSubName "Symbol") "Cons" $ T.unlines
   , "cannot be empty."
   ]
 
+symbolBreakOn :: Declaration
+symbolBreakOn = primClassOf (P.primSubName "Symbol") "BreakOn" $ T.unlines
+  [ "Compiler solved type class for breaking a symbol into its head and tail"
+  , "using the whole symbol and a breaking symbol, or for joining together the"
+  , "head and tail with the breaking symbol."
+  , "Returns empty strings when breaking on a Symbol that is not matched."
```

Comment:
Why not just not solve the constraint in this case?

Reply:
But how would I handle cases where I optionally have something? Like a `:` that might be in a given Symbol? At least this way I can chain match on the second (or Fail explicitly on empty)

Reply:
Something equivalent to a type-level `Maybe (Tuple Symbol Symbol)`?
```
foreign import kind BreakOn
foreign import data Broke :: Symbol -> Symbol -> BreakOn
foreign import data NotBroke :: BreakOn
```
If someone wants it to fail instance selection they can explicitly require `Broke first second`.

Reply:
Right, but I don't know if I want yet another kind for a class where I mostly want to use something vs ""

Reply:
Another alternative is a type-level "contains" relation which you could case on, and which could use the existing type-level boolean kit.

Reply:
I don't know what you mean, could you elaborate?

Reply:
I'm getting at the fact that `BreakOn` is only really necessary because we don't have polykinds, so we have a special-case version of `Maybe` in `Prim` for this one operation. An alternative, if you need to know if the split will be successful, is to implement something like

```purescript
class SymbolContains (sym :: Symbol) (pat :: Symbol) (result :: Boolean) | sym pat -> result
```

which uses the existing type-level `Boolean`.

Reply:
Oh, now I see. I'll try doing this tomorrow and make the non-break case fail